include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -480,12 +494,26 @@ MultigridPreconditionerBase<dim, Number>::do_initialize_dof_handler_and_constrai
       // setup dof_handler: create dof_handler...
       auto dof_handler = new dealii::DoFHandler<dim>(*tria);
       // ... create FE and distribute it
-      if(level.is_dg)
-        dof_handler->distribute_dofs(
-          dealii::FESystem<dim>(dealii::FE_DGQ<dim>(level.degree), n_components));
-      else
-        dof_handler->distribute_dofs(
-          dealii::FESystem<dim>(dealii::FE_Q<dim>(level.degree), n_components));
+      // TODO: following if-else is not used becuade of the above
+      // AssertThrow(tria->all_reference_cells_are_hyper_cube(),(..))
```

Comment:
I guess the assert and this comment can be remove once https://github.com/exadg/exadg/pull/295 is merged.

Reply:
No, because this else branch is only for local smoothing multigrid and the changes in #295 are for global coarsening multigrid.

I wasn't sure if I should keep the if-else in the following lines and that is why I added the TODO.

Reply:
removed it in 00b2424c3fc15df281a8464dad83bf252f57f759!